src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -96,8 +100,16 @@ internal TokenCache(IServiceBundle serviceBundle)
         internal void SetServiceBundle(IServiceBundle serviceBundle)
         {
             ServiceBundle = serviceBundle;
-            TokenCacheAccessor = new TelemetryTokenCacheAccessor(ServiceBundle.TelemetryManager, ServiceBundle.PlatformProxy.CreateTokenCacheAccessor());
+            var baseTokenCacheAccessor = ServiceBundle.PlatformProxy.CreateTokenCacheAccessor();
+            TokenCacheAccessor = new TelemetryTokenCacheAccessor(ServiceBundle.TelemetryManager, baseTokenCacheAccessor);
             LegacyCachePersistence = ServiceBundle.PlatformProxy.CreateLegacyCachePersistence();
+            _defaultTokenCacheBlobStorage = ServiceBundle.PlatformProxy.CreateTokenCacheBlobStorage();
+
+            if (!(_defaultTokenCacheBlobStorage is NullTokenCacheBlobStorage) &&
```

Comment:
instead of explicit type checking could we add some bool methods onto the interface that we check for behaviors?  so then we can unit test with those behaviors and be more exlpicit in the contract?

Reply:
Upon further testing, it may well be that we can combine any of these, so removing check.